<?xml version="1.0"?>
<launch>

  <arg name="use_camera" default="true"/>

  <group if="$(arg use_camera)">
    <!-- Start camera -->
    <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen" >
      <param name="video_device" value="/dev/video0" />
      <param name="image_width" value="640" />
      <param name="image_height" value="480" />
      <param name="pixel_format" value="yuyv" />
      <param name="camera_frame_id" value="usb_cam" />
      <param name="io_method" value="mmap"/>
    </node>
  </group>

  <group unless="$(arg use_camera)">
    <node name="fake_image_publisher" pkg="detection_core" type="fake_image_publisher.py" output="screen">
    </node>
  </group>

  <!-- Begin model inference -->
  <node name="detection" pkg="detection_core" type="detection.py" output="screen">
  </node>

  <!-- Show visualization -->
  <node name="image_view" pkg="image_view" type="image_view" respawn="false" output="screen">
    <remap from="image" to="/image_detection"/>
    <param name="autosize" value="true" />
  </node>

</launch>
